samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -141,6 +143,9 @@ object SamzaContainer extends Logging {
     val systemConfig = new SystemConfig(config)
     val containerModel = jobModel.getContainers.get(containerId)
     val containerName = "samza-container-%s" format containerId
+    val jobName = config.getName.getOrElse(throw new ConfigException("Missing required config: job.name"))
```

Comment:
Does config.getName already check for app.name first? The exception message may also be misleading.

Reply:
After https://github.com/apache/samza/pull/920/files 
the planner enforces job and app name to be the same, 
so yes, by the time we land in Samza Container we would've already check both configs.